relayer/crates/starknet-integration-tests/src/impls/fork.rs

```diff
@@ -0,0 +1,554 @@
+        relayer_wallet_2: driver.starknet_chain_driver.relayer_wallet_2.clone(),
+        user_wallet_a: driver.starknet_chain_driver.user_wallet_a.clone(),
+        user_wallet_b: driver.starknet_chain_driver.user_wallet_b.clone(),
+    };
+
+    let forked_cosmos_chain_driver = CosmosChainDriver {
+        chain: forked_chain_b,
+        chain_command_path: driver.cosmos_chain_driver.chain_command_path.clone(),
+        chain_node_config: fork_chain_node_config,
+        genesis_config,
+        chain_processes,
+        validator_wallet: driver.cosmos_chain_driver.validator_wallet.clone(),
+        relayer_wallet: driver.cosmos_chain_driver.relayer_wallet.clone(),
+        user_wallet_a: driver.cosmos_chain_driver.user_wallet_a.clone(),
+        user_wallet_b: driver.cosmos_chain_driver.user_wallet_b.clone(),
+        wallets: driver.cosmos_chain_driver.wallets.clone(),
+    };
+
+    Ok(StarknetTestDriver {
+        relay_driver_a_b: driver.relay_driver_a_b.clone(),
+        relay_driver_b_a: driver.relay_driver_b_a.clone(),
+        starknet_chain_driver: forked_starknet_chain_driver,
+        cosmos_chain_driver: forked_cosmos_chain_driver,
+        client_id_a: driver.client_id_a.clone(),
+        client_id_b: driver.client_id_b.clone(),
+        connection_id_a: driver.connection_id_a.clone(),
+        connection_id_b: driver.connection_id_b.clone(),
+        channel_id_a: driver.channel_id_a.clone(),
+        channel_id_b: driver.channel_id_b.clone(),
+        port_id_a: driver.port_id_a.clone(),
+        port_id_b: driver.port_id_b.clone(),
+        create_client_payload_options_a: driver.create_client_payload_options_a.clone(),
+        create_client_payload_options_b: driver.create_client_payload_options_b.clone(),
+        create_client_message_options_a: (),
+        create_client_message_options_b: (),
+        recover_client_payload_options_a:
+            hermes_starknet_chain_components::impls::StarknetRecoverClientPayload,
+        recover_client_payload_options_b: driver.recover_client_payload_options_b.clone(),
+    })
+}
+
+async fn fork_starknet_chain(
+    driver: &StarknetTestDriver,
+) -> Result<StarknetTestDriver, HermesError> {
+    let runtime = driver.starknet_chain_driver.chain.runtime.clone();
+    let chain_home_dir = driver.starknet_chain_driver.chain_store_dir.clone();
+    let chain_node_config = driver.starknet_chain_driver.node_config.clone();
+
+    let starknet_pid = driver
+        .starknet_chain_driver
+        .chain_processes
+        .first()
+        .expect("Failed to retrieve Starknet chain process")
+        .id()
+        .expect("failed to reterieve Starknet chain process ID");
+
+    // Stop full node
+    // `kill` is used here instead of `Child::kill()` as the `kill()` method requires
+    // the child process to be mutable.
+    runtime
+        .exec_command(
+            &PathBuf::from("kill".to_string()),
+            &["-s", "9", &starknet_pid.to_string()],
```

Comment:
```suggestion
            &["-s", "KILL", &starknet_pid.to_string()],
```

for macos compatibility.